### ISSUE TITLE
.NET P5: The System.ComponentModel.Composition NuGet doesn't work

### ISSUE THREAD

**Issue by rolfbjarne** [xamarin/xamarin-macios#11710]
### Steps to Reproduce

Reference the System.ComponentModel.Composition NuGet in a project and use it:

```xml
<?xml version="1.0" encoding="utf-8"?>
<Project Sdk="Microsoft.NET.Sdk">
  <PropertyGroup>
    <TargetFramework>net6.0-ios</TargetFramework>
    <RuntimeIdentifier>iossimulator-x64</RuntimeIdentifier>
    <OutputType>Exe</OutputType>
  </PropertyGroup>
  <ItemGroup>
    <PackageReference Include="System.ComponentModel.Composition" Version="6.0.0-preview.4.21253.7" />
  </ItemGroup>
</Project>
```

"dotnet restore" will resolve to the `xamarinios10` assembly, as can be seen in the generated project.assets.json file:

```json
{
  "version": 3,
  "targets": {
    "net6.0-ios13.6": {
      "System.ComponentModel.Composition/5.0.0": {
        "type": "package",
        "compile": {
          "ref/xamarinios10/_._": {}
        },
        "runtime": {
          "lib/xamarinios10/_._": {}
        }
      }
    },
```

The `_._` file means "use the file shipped with Xamarin.iOS"

Previously the package restore would use the `netstandard2.0` assembly:

```json
{
  "version": 3,
  "targets": {
    "net6.0-ios13.6": {
      "System.ComponentModel.Composition/4.7.0": {
        "type": "package",
        "compile": {
          "ref/netstandard2.0/System.ComponentModel.Composition.dll": {}
        },
        "runtime": {
          "lib/netcoreapp2.0/System.ComponentModel.Composition.dll": {}
        }
      },
```

The behavior can be seen here: https://github.com/xamarin/xamarin-macios/pull/11672#commitcomment-51360284

~This is presumably by design: https://github.com/dotnet/designs/blob/main/accepted/2021/net6.0-tfms/net6.0-tfms.md#compatibility-rules, but this needs to be verified. If it is by design, all NuGets that do the same thing (point to the assembly shipped with Xamarin.iOS) will have to be updated to provide an asset with a higher priority than `xamarinios10`.~

Current plan: update the TFM fallback rules (https://github.com/dotnet/designs/pull/222).

## Notes

* Both .NET 6 P4 and .NET 6 P6 work, while .NET 6 P5 doesn't. This is possibly because P5 has fixes P6 doesn't yet have.
* This applies to the System.Json NuGet as well, so it probably applies to all NuGets that point to assemblies shipped with Xamarin.iOS.

---

To be reverted once fixed: https://github.com/xamarin/xamarin-macios/pull/11672/commits/99c7763aab5f243cce66940b924ceba154efb008

**Comment by akoeplinger** [comment]
System.ComponentModel.Composition will be fixed in 6.0 preview6, but System.Json [will not be fixed](https://github.com/dotnet/designs/pull/222#issuecomment-861617052) so you'll need to remove any dependencies on it.

**Comment by rolfbjarne** [comment]
The System.ComponentModel.Composition NuGet has `net6.0` assets now:

```shell
$ unzip -l system.componentmodel.composition.6.0.0-preview.6.21352.12.nupkg
Archive:  system.componentmodel.composition.6.0.0-preview.6.21352.12.nupkg
  Length      Date    Time    Name
---------  ---------- -----   ----
      527  07-02-2021 22:04   _rels/.rels
     2901  07-02-2021 22:04   System.ComponentModel.Composition.nuspec
      438  07-02-2021 22:04   buildTransitive/netcoreapp2.0/System.ComponentModel.Composition.targets
        0  06-11-2021 19:19   buildTransitive/netcoreapp3.1/_._
     7006  06-11-2021 19:19   Icon.png
        0  06-11-2021 19:19   lib/MonoAndroid10/_._
        0  06-11-2021 19:19   lib/MonoTouch10/_._
        0  06-11-2021 19:19   lib/net461/_._
   278640  07-02-2021 21:59   lib/net6.0/System.ComponentModel.Composition.dll
   297260  10-09-2020 20:10   lib/net6.0/System.ComponentModel.Composition.xml
   280176  07-02-2021 21:59   lib/netcoreapp3.1/System.ComponentModel.Composition.dll
   297260  10-09-2020 20:10   lib/netcoreapp3.1/System.ComponentModel.Composition.xml
    92784  07-02-2021 21:59   lib/netstandard2.0/System.ComponentModel.Composition.dll
   297260  10-09-2020 20:10   lib/netstandard2.0/System.ComponentModel.Composition.xml
        0  06-11-2021 19:19   lib/uap10.0.16299/_._
        0  06-11-2021 19:19   lib/xamarinios10/_._
        0  06-11-2021 19:19   lib/xamarinmac20/_._
        0  06-11-2021 19:19   lib/xamarintvos10/_._
        0  06-11-2021 19:19   lib/xamarinwatchos10/_._
     1139  07-02-2021 21:45   LICENSE.TXT
        0  06-11-2021 19:19   ref/MonoAndroid10/_._
        0  06-11-2021 19:19   ref/MonoTouch10/_._
        0  06-11-2021 19:19   ref/net461/_._
    44656  07-02-2021 21:50   ref/net6.0/System.ComponentModel.Composition.dll
   297260  10-09-2020 20:10   ref/net6.0/System.ComponentModel.Composition.xml
    45168  07-02-2021 21:49   ref/netcoreapp3.1/System.ComponentModel.Composition.dll
   297260  10-09-2020 20:10   ref/netcoreapp3.1/System.ComponentModel.Composition.xml
    46208  07-02-2021 21:49   ref/netstandard2.0/System.ComponentModel.Composition.dll
   297260  10-09-2020 20:10   ref/netstandard2.0/System.ComponentModel.Composition.xml
        0  06-11-2021 19:19   ref/uap10.0.16299/_._
        0  06-11-2021 19:19   ref/xamarinios10/_._
        0  06-11-2021 19:19   ref/xamarinmac20/_._
        0  06-11-2021 19:19   ref/xamarintvos10/_._
        0  06-11-2021 19:19   ref/xamarinwatchos10/_._
    47629  07-02-2021 21:45   THIRD-PARTY-NOTICES.TXT
      782  07-02-2021 22:04   [Content_Types].xml
     1685  07-02-2021 22:04   package/services/metadata/core-properties/2387831c0f8c432d912d0b2d2ff24e6f.psmdcp
    18702  07-14-2021 06:52   .signature.p7s
---------                     -------
  2652001                     38 files
```

I've added a note about the deprecated packages (such as System.Json) here: #10574 for future release notes.
